### DISCUSSION TITLE
Displaced jets

### DISCUSSION THREAD

**Discussion by matiasmantinan** [HSF/phoenix#691] [Q&A]
Hi, Is there a way to create a jet that is not center in the origin? I'm looking at the event data format (https://github.com/HSF/phoenix/blob/main/guides/developers/event_data_format.md) and I don't see any option to record the location of the jets.

**Comment by EdwardMoyse** [comment]
Sorry - I just saw this! No, we don't support this at the moment but it should be pretty easy to add. Are you still interested?

**Reply by matiasmantinan** [reply]
Yes, I think it would be a really nice feature to have! I can try to add it myself if you point to the pieces of code I need to modify.

**Reply by EdwardMoyse** [reply]
Great!

The code that draws the jet is here:
https://github.com/HSF/phoenix/blob/f58715f1b3b61b079a63bc403c690de4453fa03a/packages/phoenix-event-display/src/loaders/objects/phoenix-objects.ts#L201

You can see that it takes `jetParams` so the first step would be to conditionally handle "origin" coordinates and draw the jet accordingly (i.e. it should check if the x,y,z origin is in the parameters, and only handle a displaced jet if so). 

To start with, you could hardcode this, but ultimately it would need to come from the calling code, so e.g.
* cms-loader
* edm4hep-json-loader
* jivexml-loader
* phoenix-loader

Also, the documentation would need updating.
